pkg/auth/tokengenerator.go

```diff
@@ -0,0 +1,44 @@
+package auth
+
+import (
+	"fmt"
+)
+
+// TokenGenerator allows for the creation of tokens
+type TokenGenerator interface {
+	// GetToken returns a token which can be used for
+	// authentication and communication from node to node.
+	GetToken(opts *Options) (string, error)
+
+	// Issuer returns the token issuer for this generator necessary
+	// for registering the authenticator in the SDK.
+	Issuer() string
+
+	// GetAuthenticator returns an authenticator for this issuer used by the SDK
+	GetAuthenticator() (*JwtAuthenticator, error)
```

Comment:
I thought the return value should be auth.Authenticator, right?

Reply:
This has always been JwtAuthenticator:
https://github.com/portworx/porx/blob/sdk-master/pkg/auth/auth.go#L16